zha/zigbee/device.py

```diff
@@ -868,6 +869,81 @@ def _is_entity_removed_by_quirk(self, entity: PlatformEntity) -> bool:
 
         return False
 
+    def _apply_entity_metadata_changes(self, entity: PlatformEntity) -> None:
+        """Apply entity metadata changes from quirks v2."""
+        if self.quirk_metadata is None:
+            _LOGGER.debug("No quirk metadata for device %s", self)
+            return
+
+        _LOGGER.debug(
+            "Processing entity %s for metadata changes. Quirk metadata has %d change entries",
+            entity,
+            len(self.quirk_metadata.changed_entity_metadata),
+        )
```

Comment:
Even though this is debug level, I could see this become a bit spammy, since one of the messages will be printed for **every entity**, even if there's no quirk applied or no change entries for a quirk. 

But yeah, it's debug level, so it's *probably* fine.